src/libstd/sys/unix/thread.rs

```diff
@@ -326,11 +326,23 @@ pub mod guard {
             // Reallocate the last page of the stack.
             // This ensures SIGBUS will be raised on
             // stack overflow.
-            let result = mmap(stackaddr, PAGE_SIZE, PROT_NONE,
-                              MAP_PRIVATE | MAP_ANON | MAP_FIXED, -1, 0);
+            if cfg!(target_os = "netbsd") {
```

Comment:
Can you add a comment with an abridged version of the PR description, mentioning the Issue #? That way people can track the reason for this without having to use `git blame` to fish out why netbsd is being singled out here.

Reply:
It affects anyone who is enforcing PaX MPROTECT in strict mode. Strict mode means not only you are enforcing W^X (i.e. you can't map a segment both writable and executable at the same time), but also that you can't use mprotect(2) to augment (add) permissions on an already mapped page (that did not have those permissions during mapping time). I.e. can't toggle between R-X<->RW- using mprotect(2). I am not aware of other OS's doing this at this time, but the change to do the mmap(PROT_READ|PROT_WRITE)+mprotect(PROT_NONE) for the guard page should work on every OS, so it doesn't really need to be conditionalized. I would put a comment why this is the case though (why we are going doing this).